### PR TITLE
TF-3083 PublicAsset Partial update

### PR DESCRIPTION
**Pull request by tddang-linagora**
## Issue
- #3083

### CODE REVIEW

File: lib/features/public_asset/data/network/public_asset_api.dart [hunk 1]
```diff
@@ -29,6 +31,20 @@ class PublicAssetApi {
     return MapEntry(publicAsset.id!, patchObject);
   }
 
+  MapEntry<PublicAssetId, PatchObject> _toPartialPatchObjectMapEntry(
+    PublicAssetId publicAssetId,
+    UpdatingIdentityIds updatingIdentityIds,
+  ) {
+    assert(
+      updatingIdentityIds.values.where((value) => value == false).isEmpty,
+      'All updating identity id values must be true or null'
```
Comment (dab246) [inline lib/features/public_asset/data/network/public_asset_api.dart:40]: Where is string?

Comment (tddang-linagora) [reply lib/features/public_asset/data/network/public_asset_api.dart:40]: What do you mean?

Comment (dab246) [reply lib/features/public_asset/data/network/public_asset_api.dart:40]: Why is this string there? Where did it come from?

Comment (tddang-linagora) [reply lib/features/public_asset/data/network/public_asset_api.dart:40]: It is there for the developers only. As is says, that `PublicAsset/set update` request only accept `true` or `null` value.

File: lib/features/public_asset/data/network/public_asset_api.dart [hunk 2]
```diff
@@ -29,6 +31,20 @@ class PublicAssetApi {
     return MapEntry(publicAsset.id!, patchObject);
   }
 
+  MapEntry<PublicAssetId, PatchObject> _toPartialPatchObjectMapEntry(
+    PublicAssetId publicAssetId,
+    UpdatingIdentityIds updatingIdentityIds,
+  ) {
+    assert(
+      updatingIdentityIds.values.where((value) => value == false).isEmpty,
```
Comment (dab246) [inline lib/features/public_asset/data/network/public_asset_api.dart:39]:
```suggestion
      updatingIdentityIds.values.every((value) => value != false),
```

**Comment by dab246** [comment]
- [ ] Please rebase it

File: lib/features/public_asset/data/network/public_asset_api.dart (hunk 2 above)
Comment (tddang-linagora) [reply lib/features/public_asset/data/network/public_asset_api.dart]: Done

File: test/features/public_asset/data/network/public_asset_api_test.dart [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+            SetPublicAssetResponse.deserialize);
+      
+      // assert
+      expect(response?.updated, equals({publicAsset.id: null}));
+      await expectLater(
+        publicAssetApi.partialUpdatePublicAssets(
+          session,
+          accountId,
+          mapPublicAssetIdToUpdatingIdentityIds: mapPublicAssetIdToUpdatingIdentityIds),
+        completes);
```
Comment (hoangdat) [inline test/features/public_asset/data/network/public_asset_api_test.dart:111]: what is it?

File: lib/features/public_asset/data/network/public_asset_api.dart [hunk 4]
```diff
@@ -134,4 +150,29 @@ class PublicAssetApi {
       throw const CannotUpdatePublicAssetException();
     }
   }
+
+  Future<void> partialUpdatePublicAssets(
+    Session session,
+    AccountId accountId,
+    {required Map<PublicAssetId, UpdatingIdentityIds> mapPublicAssetIdToUpdatingIdentityIds}
```
Comment (hoangdat) [inline lib/features/public_asset/data/network/public_asset_api.dart:157]: can you test the case with multiple Entries in this map?

File: test/features/public_asset/data/network/public_asset_api_test.dart (hunk 3 above)
Comment (tddang-linagora) [reply test/features/public_asset/data/network/public_asset_api_test.dart:111]: `completes` is to check if a `Future<void>` function completes without throwing any exception. See more [here](https://api.flutter.dev/flutter/package-matcher_expect/completes.html)

File: lib/features/public_asset/data/network/public_asset_api.dart (hunk 4 above)
Comment (tddang-linagora) [reply lib/features/public_asset/data/network/public_asset_api.dart:157]: Test updated for multiple entries with different update types

**Review by hoangdat — APPROVED** [review]

**Comment by hoangdat** [comment]
- [ ] please rebase it

**Review by dab246 — APPROVED** [review]

**Pull request merged**